ragna/assistants/_anthropic.py

```diff
@@ -18,26 +19,38 @@ def _extra_requirements(cls) -> list[Requirement]:
     def display_name(cls) -> str:
         return f"Anthropic/{cls._MODEL}"
 
-    def _instructize_prompt(self, prompt: str, sources: list[Source]) -> str:
-        # See https://docs.anthropic.com/claude/docs/introduction-to-prompt-design#human--assistant-formatting
+    def _instructize_system_prompt(self, sources: list[Source]) -> str:
+        # See https://docs.anthropic.com/claude/docs/system-prompts
+        # See https://docs.anthropic.com/claude/docs/long-context-window-tips#tips-for-document-qa
         instruction = (
-            "\n\nHuman: "
-            "Use the following pieces of context to answer the question at the end. "
-            "If you don't know the answer, just say so. Don't try to make up an answer.\n"
+            "I'm going to give you a document."
+            "Read the document carefully, because I'm going to ask you a question about it."
+            "If you don't know the answer, just say so. Don't try to make up an answer."
+            "Only use the included documents below to generate the answer.\n"
         )
-        instruction += "\n\n".join(source.content for source in sources)
-        return f"{instruction}\n\nQuestion: {prompt}\n\nAssistant:"
+        # See https://docs.anthropic.com/claude/docs/long-context-window-tips#structuring-long-documents
+        documents = ElementTree.Element("documents")
+        for idx, source in enumerate(sources, start=1):
+            doc_elmnt = ElementTree.SubElement(
+                documents,
+                "document",
+                attrib={"index": str(idx)},
+            )
+            ElementTree.SubElement(doc_elmnt, "id").text = source.id
+            ElementTree.SubElement(doc_elmnt, "document_content").text = source.content
```

Comment:
I started rolling my own XML, but I felt it was more robust to use a built-in library. 

Also, the tagging scheme I have matches [the documnetation](https://docs.anthropic.com/claude/docs/long-context-window-tips#structuring-long-documents).

Reply:
Sure, it is more robust to use a library, but what I'm saying here is that you include a ton of information that is not needed by the LLM and might be even detrimental for the response it gives:

1. While we can internally label the sources as "documents", the LLM should never refer to them in this way in the response. From a user perspective a "document" is one file that they uploaded, but here we only deal with chunks of these documents. Meaning, if the LLM responds with something like "According to document 1, ...", this will only lead to confusion. Thus, we shouldn't number the documents here and give the LLM a way to refer to them.
2. Similar to the point above, the ID is not useful in any way. This is a purely internal ID that the user will never see. Meaning, on top of the point above, the prompt now has more tokens for no reason.

Lastly, I also think we can diverge from the actual HTML tags that they have used in their documentation. They only speak about wrapping everything in HTML tags, but not that one should use specific ones. Thus, I'm ok with wrapping the whole block in `<documents>` and each individual in `<document>`. For more, I want to see some evidence that this actually improves things.

And for this simple scheme, we don't need to pull in an extra library to handle the wrapping. My concern here is maintainability. Looking at your code, I cannot see at a glance what is happening. Unless this improves something, we shouldn't prefer that over a simpler solution.